File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Order extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id', 'customer_id', 'customer_name', 'product_type', 'color',
        'size', 'quantity', 'delivery_date', 'status', 'notes',
        'unit_price', 'unit_cost',
    ];

    protected $casts = [
        'delivery_date' => 'date',
        'quantity'      => 'integer',
        'unit_price'    => 'decimal:2',
        'unit_cost'     => 'decimal:2',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function customer()
    {
        return $this->belongsTo(Customer::class);
    }

    public function productionOrder()
    {
        return $this->hasOne(ProductionOrder::class);
    }
}

File: backend-dotnet/Models/Order.cs
namespace TexTrack.Api.Models;

public class Order
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public long? CustomerId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string ProductType { get; set; } = string.Empty;
    public string Color { get; set; } = string.Empty;
    public string Size { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public DateOnly DeliveryDate { get; set; }
    public string Status { get; set; } = "Pending";
    public string? Notes { get; set; }
    public decimal? UnitPrice { get; set; }
    public decimal? UnitCost { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User? User { get; set; }
    public Customer? Customer { get; set; }
    public ProductionOrder? ProductionOrder { get; set; }
}
